### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by jakehein**
### All Submissions

* [ ] Can you post a screenshot of your changes (if applicable)?

### New Feature Submissions

* [ ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

Working on Modularizing Store for Combat specific functions/data. This is part of #1284

### Testing

Has the code been tested, or does it need double checking?

### CODE REVIEW

**Comment by kevinunger** [comment]
You are using the getter `getWeaponDurability()` as `combat/getWeaponDurability` in some place, but the getter isn't in the combat module.
I'd say we leave that in the 'general' store and just remove the `combat/` where it is used.

**Comment by kevinunger** [comment]
Some actions/mutations like `updateXpRewards`, `updateCharacterStamina`aren't properly namespaced where they are used. The best, but tedious way to find the all references is to search for all occurences of each function in /store/combat.ts one-by-one in the whole codebase. These also appear as errors or warnings in the console.

File: frontend/src/views/Combat.vue [hunk 1]
```diff
@@ -349,19 +461,19 @@ export default {
...
-    getElementAdvantage(playerElement, enemyElement) {
+    getElementAdvantage(playerElement: number, enemyElement: number) {
       if ((playerElement + 1) % 4 === enemyElement) return 1;
       if ((enemyElement + 1) % 4 === playerElement) return -1;
       return 0;
     },
 
     // for enemy card animaton
-    enter(el, done) {
+    enter(el: any, done: any) {
```
Comment (kevinunger) [inline frontend/src/views/Combat.vue:476]: this could be `el:HTMLElement`

File: frontend/src/views/Combat.vue [hunk 2]
```diff
@@ -370,13 +482,13 @@ export default {
         delay: el.dataset.index * 0.1
       });
     },
-    beforeEnter(el){
+    beforeEnter(el: any){
```
Comment (kevinunger) [inline frontend/src/views/Combat.vue:485]: this could be el:HTMLElement

File: frontend/src/views/Combat.vue [hunk 3]
```diff
@@ -535,7 +649,7 @@ export default {
       }
       return x1 + x2;
     },
```
Comment (kevinunger) [inline frontend/src/views/Combat.vue:651]: We could remove this crazy regex addCommas() and just change L129 to `{{e.power.toLocaleString()}}`, where it is used.

**Comment by jakehein** [comment]
made changes, currently getting issue with state.contracts apparently not being a function. Not sure how I'm getting this error and seemingly nowhere else is. Any input would be appreciated 
image
image

**Comment by kevinunger** [comment]
@jakehein Yeah, you have to use rootState for that. 
The rootState object is the state defined in our store.ts, which has the contracts. The state is your module state of your vuex-module (saw you used that as well already; so I guess that here is just an overlooked typo)

**Review by kevinunger — APPROVED** [review]

**Pull request merged**